auth/auth.go

```diff
@@ -239,9 +274,36 @@ func (auth *Authenticator) rebuildChannels(princ Principal) (changed bool, err e
 					return changed, err
 				}
 				changed = true
+				if auth.ServerlessChannelThreshold != 0 {
+					if user, ok := princ.(User); ok {
+						channelCount := auth.channelCount(princ, user, roles, scope, collection)
+						cumulativeChannels += channelCount
+					}
+				}
+			}
+		}
+	}
+
+	// only warn/limit if the threshold is set and if we are in this function as a "user" not a role
+	if auth.ServerlessChannelThreshold != 0 && princUser != nil {
+		// Warning at 50 channels
+		princUser.GetWarnChanSync().Do(func() {
```

Comment:
This is going to not work as intended, since this only works the version first time this code is run, I'd write a test where it is over threshold the first time and then under threshold the second time, and it should only warn once.

What happens if it high (warning), then low, then high again (error)?

Reply:
I took this apporach as elsewhere we do this warning using the sync once. I don't think you can reset the sync.Once so I either keep as is to follow what is done elsewhere in the code or I remove it (but could end up with a lot of warnings in logs as result)